feat: SEO Component & Application Refactor.

ahmadawais opened the pull request:
Hey, folks! 🙌

Would love to be involved in the project. First PR here, closed the #67 since I have refactored the app and included a better reusable SEO component.


### This PR does a couple of related things:

* [x]  Adds an SEO component
* [x]  Adds SEO: OGG, Twitter, Facebook, and general metadata
* [x]  Adds featured image metadata for future usage with social sites
* [x]  Refactors entire Layout file with SEO component to keep things civil
* [x]  Adds auto-generating canonical-urls for the sake of SEO with a plugin
* [x]  Refactors Header (alt tag for this site will always be static === more performant)
* [x]  Refactors the `learn` & `404` pages to include `title`, `description`, and `author`.



### Tests ran

* [x]  `npm start` — without errors
* [x]  `npm run build` — without errors

Looking forward, peace! ✌️

CC: @MylesBorins kindly check this PR instead.

MylesBorins commented:
/gcbrun

eventual staging example will be on https://storage.googleapis.com/staging.nodejs.dev/b3322e4/index.html

ahmadawais commented:
Staging example looks fine to me. Let me know if there's anything to be done here. Looking forward, peace! ✌️

ahmadawais commented:
@LaRuaNa That's true. In my projects, I have a base config file which avoids all the repetition and extra GraphQL queries (which are negligible). I can do a second pass on this if that's the direction we want to go in.

ahmadawais commented:
@LaRuaNa I've just added a config file and changed all the instances of information inside the `seo` and `gatsby-config` file to use that config file instead of extra GraphQL queries. 

Also, resolved several merge conflicts. This is a big PR. Can you kindly look into this and merge it timely, so we can avoid more conflicts. Here if you need any help or have any questions.

Looking forward, peace! ✌️

LaRuaNa commented:
@ahmadawais nice, but fails to build can you check that please :]

ahmadawais commented:
Everything was fine at my end until I saw the `yarn.lock` file needed to be updated. And then an old test was failing so I rewrote the test since the header is now different. Also udpated the snapshots `./node_modules/.bin/jest --updateSnapshot`

All checks passed now. Please, check!

ollelauribostrom commented:
Looks good to me 👍

ahmadawais commented:
Fixed two new merge conflicts. 😭

amiller-gh commented:
Taking a look now @ahmadawais 🙂 Sorry for the delay!

amiller-gh commented on src/pages/learn.tsx:
```diff
@@ -1,37 +1,38 @@
-import React from 'react';
```
Where'd all the semicolons go? 🙂

MylesBorins replied:
prettier

ahmadawais commented:
@amiller-gh I'm looking forward to the meeting next week would love to take responsibility for choosing formatting standards and apply that formatting everywhere.

Currently, my VSCode is set up to follow the present `.prettierrc` file which is why all the formatting changes. 

My personal preference is the settings below but will discuss that in the next meeting:

```json
{
	"semi": true,
	"singleQuote": true,
	"trailingComma": "es5",
	"useTabs": true,
	"bracketSpacing": true,
	"printWidth": 120,
	"tabWidth": 4
}

```

Thanks for merging the PR. Wohoo! 🎉

LaRuaNa commented:
> Fixed two new merge conflicts. 😭

@ahmadawais  I'm really sorry for that mate. My bad :[ Thanks for working on this 🙏

ahmadawais commented:
@laruana no worries at all. ☝👍👌 
Thank you!
